Formula/ghc@8.6.rb

```diff
@@ -20,14 +20,22 @@ class GhcAT86 < Formula
   keg_only :versioned_formula
 
   depends_on "python@3.9" => :build
-  depends_on "sphinx-doc" => :build
   depends_on arch: :x86_64
 
-  resource "gmp" do
-    url "https://ftp.gnu.org/gnu/gmp/gmp-6.1.2.tar.xz"
-    mirror "https://gmplib.org/download/gmp/gmp-6.1.2.tar.xz"
-    mirror "https://ftpmirror.gnu.org/gmp/gmp-6.1.2.tar.xz"
-    sha256 "87b565e89a9a684fe4ebeeddb8399dce2599f9c9049854ca8c0dfbdea0e21912"
+  uses_from_macos "m4" => :build
+  uses_from_macos "ncurses"
+
+  on_macos do
+    resource "gmp" do
+      url "https://ftp.gnu.org/gnu/gmp/gmp-6.1.2.tar.xz"
+      mirror "https://gmplib.org/download/gmp/gmp-6.1.2.tar.xz"
+      mirror "https://ftpmirror.gnu.org/gmp/gmp-6.1.2.tar.xz"
+      sha256 "87b565e89a9a684fe4ebeeddb8399dce2599f9c9049854ca8c0dfbdea0e21912"
+    end
+  end
+
+  on_linux do
+    depends_on "gmp" => :build
```

Comment:
Does the `linuxbrew-core` formula actually use `gmp` formula or does it statically link the same resource as macOS? The upstream formula is still building/installing a static `gmp` in the `ghc@8.6` formula https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/ghc%408.6.rb#L69-L81

Current linkage failure with `gmp` formula:
```
==> brew linkage --test ghc@8.6
==> FAILED
Unwanted system libraries:
  /usr/lib/x86_64-linux-gnu/libgmp.so.10
```

Reply:
`gmp` is a dependency here for the boostrap binaries, which apparently are not statically linked to `libgmp`.